src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectStateChangeDetector.cs

```diff
@@ -10,6 +10,7 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Razor.Language.Components;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.VisualStudio.Threading;
```

Comment:
So this is the magic that's enabling you to `await` a `TaskScheduler`. Typically that wouldn't be possible, and in our language server we don't depend on Visual Studio at all which would make depending on the `ForegroundDispatcher`'s `SwitchToForeground` method less seamless 😄. I know I suggested to have a `SwitchToForeground` method but now I'm thinking maybe a `RunOnForegroundAsync<TParam>(Action<TParam> action)` may be a better route to take so we fully understand what we're running on the foreground (since it'll be encapsulated in the `Task`) and what we're not. That way you can utilize `Task.Factory.StartNew` under the covers too.